src/odemis/acq/stream/_base.py

```diff
@@ -787,6 +835,172 @@ def _image_thread(wstream):
+                tiles_column.append(tile)
+            tiles.append(tiles_column)
+
+        return img.mergeTiles(tiles)
+
+    def _getMergedRawImage(self, z):
+        """
+        Returns the merged image based on z and .rect, using the raw tiles (not projected)
+        z (int): Zoom level index
+        return (DataArray): The merged image
+        """
+        # TODO remove duplicate code from _getMergedImage
+        # calculates the size of the merged image
+        width_zoomed = self.raw.shape[1] / (2 ** z)
+        height_zoomed = self.raw.shape[0] / (2 ** z)
+        # calculates the number of tiles on both axes
+        num_tiles_x = int(math.ceil(width_zoomed / self.raw.tile_shape[1]))
+        num_tiles_y = int(math.ceil(height_zoomed/ self.raw.tile_shape[0]))
+
+        tiles = []
+        for x in range(num_tiles_x):
+            tiles_column = []
+            for y in range(num_tiles_y):
+                tile = self.raw.getTile(x, y, z)
+                tiles_column.append(tile)
+            tiles.append(tiles_column)
+
+        return img.mergeTiles(tiles)
+
+    def _getProjectedTile(self, x, y, z, previous_cache):
+        """
+        Get a tile from a DataArrayShadow. Uses cache.
+        x (int): The X coordinate of the tile
+        y (int): The Y coordinate of the tile
+        z (int): The zoom level where the tile is
+        previous_cache (dictionary): The cache from the last iteration of _updateImage
+        return (DataArray): The projected tile
+        """
+        # the key of the tile on the cache
+        tile_key = "%d-%d-%d" % (x, y, z)
+        # if the tile has been already cached, read it from the cache
+        if tile_key in previous_cache:
+            tile = previous_cache[tile_key]
+        else:
+            # The tile was not cached. Read it, and insert it on the cache
+            tile = self.raw.getTile(x, y, z)
+            tile = self._projectTile(tile)
+
+        # cache the tile
+        self._projectedTilesCache[tile_key] = tile
+        return tile
+
+    def _projectTile(self, tile):
+        """
+        Project the tile
+        tile (DataArray): Raw tile
+        return (DataArray): Projected tile
+        """
+        return self._projectXY2RGB(tile, self.tint.value)
+
+    def _getTilesFromSelectedArea(self):
+        """
+        Get the tiles inside the region defined by .rect and .mpp
+        return (DataArray): The image
+        """
+        z = self._zFromMpp()
+        rect = self._rectWorldToPixel(self.rect.value)
+        # convert the rect coords to tile indexes
+        rect = [l / (2 ** z) for l in rect]
+        rect = [int(math.floor(l / self.raw.tile_shape[0])) for l in rect]
+        x1, y1, x2, y2 = rect
+        curr_mpp = self.mpp.value
+        curr_rect = self.rect.value
+        # store the previous cache to use in this iteration
+        previous_cache = self._projectedTilesCache
+        self._projectedTilesCache = {}
+        while True:
```

Comment:
"Then, start the inner loop by set curr_mpp and curr_rect.": setting to what value? self.mpp and self.rect?